plugins/services/src/js/containers/service-debug/ServiceDebugContainer.js

```diff
@@ -48,11 +50,20 @@ class ServiceDebugContainer extends React.Component {
 
     const queue = service.getQueue();
 
+    const offers = queue.declinedOffers.offers;
+    for (let i = 0; i < offers.length; i++) {
+      const hostname = offers[i].hostname;
+      const node = MesosStateStore.getNodeFromHostname(hostname);
```

Comment:
Did you try adding this to `service` struct where it gets created? I feel like that would be a better place then "somewhere down the line" 🤔

Reply:
Or actually `DeclinedOffersTable` around line 210, so that we don't render the hostname?